### DISCUSSION TITLE
Rest Client - Workaround for Known Limitation "it is not possible to set HostnameVerifier or SSLContext"

### DISCUSSION THREAD

**Discussion by akil-rails** [quarkusio/quarkus#39749] [Q&A]
The [docs](https://quarkus.io/guides/rest-client#known-limitations) clearly state that there is a known limitation that **it is not possible to set HostnameVerifier or SSLContext**.

The docs however, do not state, what is the workaround or alternative to be used in case setting the sslContext is a requirement.

Any suggestions on where to look ?

[a comment by an automated account is not shown]

**Reply by cescoffier** [reply]
You can configure the key store and truststore, which cover the ssl context. Also, for http the highly recommended host name verifier algorithm is https. There is an attribute to ignore the verification altogether (quarkus.rest-client.extensions-api.verify-host=true or false)

**Comment by crilofer** [comment]
> You can configure the key store and truststore, which cover the ssl context. Also, for http the highly recommended host name verifier algorithm is https. There is an attribute to ignore the verification altogether (quarkus.rest-client.extensions-api.verify-host=true or false)

FYI, apparently `verify-host` is not working as exposed [here](https://github.com/quarkusio/quarkus/issues/39810#issuecomment-2029684747)

**Reply by cescoffier** [reply]
No, I commented on the issue, there is a difference better the hostname verification and trusting TLS certificates.
